### PR TITLE
[diffusion, tests, doc] feat: add Qwen-Image SFT example

### PR DESCRIPTION
**Pull request by zyd-ustc**
### What does this PR do?

This PR adds a standalone Qwen-Image SFT example for CoRT-style `t2i` / `edit` image data.

Main changes:

- Add `examples/qwen_image_sft_trainer/qwen_image_sft.py`, a PyTorch/diffusers SFT entrypoint with LoRA, FSDP, CUDA/NPU/CPU device selection, checkpoint save/resume, and CoRT-style dataset expansion.
- Add GPU and Ascend NPU LoRA launch scripts:
  - `run_qwen_image_sft_lora.sh`
  - `run_qwen_image_sft_lora_npu.sh`
- Add README documentation for supported CoRT k-turn rows, atomic rows, weights, LoRA launch, NPU launch, and checkpoints.
- Add CPU unit tests covering temporary dummy CoRT/atomic data expansion and the loss path.

This is a standalone recipe-style trainer, not a Ray/Hydra worker integration.

AI assistance was used to draft and prepare this change. The submitter has reviewed the changed files.

### Checklist Before Starting

- [x] Search for similar PRs. Query links:
  - https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+Qwen-Image+SFT
  - https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+qwen+image+sft
  - https://github.com/verl-project/verl-omni/issues?q=is%3Aissue+is%3Aopen+qwen+image+sft
- [x] Format the PR title as `[{modules}] {type}: {description}`.

### Test

```bash
PATH=/Users/zyd/miniconda3/bin:$PATH pre-commit run --files \
  examples/qwen_image_sft_trainer/README.md \
  examples/qwen_image_sft_trainer/qwen_image_sft.py \
  examples/qwen_image_sft_trainer/run_qwen_image_sft_lora.sh \
  examples/qwen_image_sft_trainer/run_qwen_image_sft_lora_npu.sh \
  tests/trainer/diffusion/test_qwen_image_sft_example_on_cpu.py

### CODE REVIEW

**Comment by SamitHuang** [comment]
Thanks for the PR. Can you attach the training results for reference?

File: examples/qwen_image_sft_trainer/qwen_image_sft.py [hunk 1]
```diff
@@ -0,0 +1,1111 @@
...
+                    else:
+                        grad_norm = torch.nn.utils.clip_grad_norm_(pipe.transformer.parameters(), args.max_grad_norm)
+                else:
+                    grad_norm = torch.tensor(0.0, device=device)
+                optimizer.step()
+                scheduler.step()
+                optimizer.zero_grad(set_to_none=True)
+
+            global_step += 1
+            if rank == 0 and global_step % args.log_freq == 0:
```
Comment (ZihaoW123) [inline examples/qwen_image_sft_trainer/qwen_image_sft.py:1082]: `global_step` increments after every micro-batch, but `optimizer.step()` is called only after accumulation completes. However, `total_training_steps`, saving, validation, logging, and termination conditions all use this micro-batch count.

With the default script setting `gradient_accumulation_steps=4` and `total_training_steps=1000`, training ends after about 250 optimizer updates—not the expected 1000. The learning rate scheduler also advances only ~250 times.

**Suggestions:**

- Decouple `micro_step` from `optimizer.step()`.

- Increment the global training step, advance the scheduler, and trigger save/validation/stop only after `optimizer.step()`.

File: examples/qwen_image_sft_trainer/qwen_image_sft.py [hunk 2]
```diff
@@ -0,0 +1,1111 @@
...
+        val_loader = DataLoader(
+            val_dataset,
+            batch_size=args.train_batch_size,
+            sampler=val_sampler,
+            num_workers=args.dataloader_num_workers,
+            collate_fn=collate_atomic_entries,
+            drop_last=False,
+        )
+
+    pipe = load_pipeline(args, dtype=dtype, device=device)
```
Comment (ZihaoW123) [inline examples/qwen_image_sft_trainer/qwen_image_sft.py:1044]: `--resume_from` does not restore model weights.

**Locations:**

- [`qwen_image_sft.py` L142](https://github.com/verl-project/verl-omni/blob/08750ac2249a02ee4e00e9b999b2dc8697fd6e94/examples/qwen_image_sft_trainer/qwen_image_sft.py#L142)
- [`qwen_image_sft.py` L953-L964](https://github.com/verl-project/verl-omni/blob/08750ac2249a02ee4e00e9b999b2dc8697fd6e94/examples/qwen_image_sft_trainer/qwen_image_sft.py#L953-L964)
- [`qwen_image_sft.py` L1044-L1051](https://github.com/verl-project/verl-omni/blob/08750ac2249a02ee4e00e9b999b2dc8697fd6e94/examples/qwen_image_sft_trainer/qwen_image_sft.py#L1044-L1051)

**Issue：**

The help text says it loads checkpoints saved by this script, but the implementation only restores the optimizer, scheduler, and step counter. The transformer model is still initialized from `model_name_or_path`. Users passing `--resume_from global_step_xxx` will train the base model weights with an old optimizer state.

**Suggestions:**

- Restore the model from `<resume_from>/transformer`, or
- Rename the argument to `--resume_trainer_state_from` and force users to point `model_name_or_path` to the checkpoint transformer when resuming.

**Comment by wtomin** [comment]
Close this PR because it's overlapped with #295 and partially coverred in related PRs. Thanks for your contributions!
